lib/grizzly/zwave_firmware.ex

```diff
@@ -0,0 +1,165 @@
+defmodule Grizzly.ZwaveFirmware do
+  @moduledoc false
+
+  require Logger
+
+  @doc """
+  Update the ZWave chip firmware if zw_programmer is present and a
+  valid GBL file is available that would up the firmware version.
+  """
+  @spec maybe_run_zwave_firmware_update(Grizzly.Options.t()) :: :ok
+  def maybe_run_zwave_firmware_update(options) do
+    with :ok <- check_zwave_programmer_exists(options),
+         :ok <- maybe_upload_zwave_firmware_update(options) do
+      :ok
+    else
+      {:error, reason} ->
+        Logger.info("[Grizzly] The hub's z-wave firmware was NOT updated: #{inspect(reason)}")
+    end
+
+    :ok
+  end
+
+  defp check_zwave_programmer_exists(options) do
+    if File.exists?(options.zw_programmer_path) do
+      :ok
+    else
+      {:error, :zw_programmer_does_not_exist}
+    end
+  end
+
+  defp maybe_upload_zwave_firmware_update(options) do
```

Comment:
```suggestion
```

It looks like this can be deleted. The Z-Wave programmer path is checked in the call to `System.cmd` so I don't think it's necessary to check again here.

Reply:
If the executable does not exists and we try to run `System.cmd` it will raise a runtime error.